### ISSUE TITLE
decimal division by 0

### ISSUE THREAD

**Issue by amanenk** [infracost/infracost#1922]
Hello. I am having problems with running infracost action after `terraform plan`. I have added some glue resources and infracost started crashing with `decimal division by 0`. I've removed my glue resources to roll everything back but it still crashes.
Below are the logs from the action:
```
Run infracost breakdown --config-file=/tmp/infracost.yml --format=json --out-file=/tmp/infracost.json
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/accessanalyzer/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/accessanalyzer/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/athena/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/athena/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/autoscaling/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/autoscaling/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/backup/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/backup/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/cloudformation/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/cloudformation/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/codepipeline/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/codepipeline/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/elasticbeanstalk/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/elasticbeanstalk/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/ecs/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/ecs/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/ec2/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/ec2/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/glue/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/glue/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/apigateway2/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/apigateway2/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=info msg="Stripped wrapper output from terraform/iot/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:48Z" level=info msg="Detected Terraform plan JSON file at terraform/iot/prod/plan.json"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:48Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:48Z" level=warning msg="No products found for aws_launch_configuration Instance usage (Linux/UNIX, on-demand, ), using 0.00"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/lambda/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/lambda/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/qldb/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/qldb/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=warning msg="No products found for aws_launch_configuration Instance usage (Linux/UNIX, on-demand, ), using 0.00"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/lightsail/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/lightsail/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/s3/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/s3/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/sagemaker/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/sagemaker/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/ses/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/ses/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/sns/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/sns/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/sqs/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/sqs/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/mq/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/mq/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/redshift/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/redshift/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/rds/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/rds/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/ssm/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/ssm/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/waf/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/waf/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/wafregional/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/wafregional/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/wafv2/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/wafv2/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/xray/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/xray/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-07-28T08:44:49Z" level=info msg="Stripped wrapper output from terraform/workspaces/prod/plan.json (to make it a valid JSON file) since setup-terraform GitHub Action was used without terraform_wrapper: false"
time="2022-07-28T08:44:49Z" level=info msg="Detected Terraform plan JSON file at terraform/workspaces/prod/plan.json"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-07-28T08:44:49Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"

Error: An unexpected error occurred

decimal division by 0
goroutine 1 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/1.18.4/x64/src/runtime/debug/stack.go:24 +0x65
main.Run.func1()
	/home/runner/work/infracost/infracost/cmd/infracost/main.go:68 +0xc6
panic({0x1c847a0, 0x2798320})
	/opt/hostedtoolcache/go/1.18.4/x64/src/runtime/panic.go:838 +0x207
github.com/shopspring/decimal.Decimal.QuoRem({0xc0000a3ee0, 0x1}, {0xc00067b0c0, 0x0}, 0x10)
	/home/runner/go/pkg/mod/github.com/shopspring/decimal@v1.3.1/decimal.go:565 +0x2c5
github.com/shopspring/decimal.Decimal.DivRound({0xc0000a3ee0?, 0x986ff8?}, {0xc00067b0c0?, 0x50?}, 0x10)
	/home/runner/go/pkg/mod/github.com/shopspring/decimal@v1.3.1/decimal.go:607 +0x56
github.com/shopspring/decimal.Decimal.Div(...)
	/home/runner/go/pkg/mod/github.com/shopspring/decimal@v1.3.1/decimal.go:552
github.com/infracost/infracost/internal/schema.(*CostComponent).UnitMultiplierHourlyQuantity(0xc000882fa0)
	/home/runner/work/infracost/infracost/internal/schema/cost_component.go:75 +0x65
github.com/infracost/infracost/internal/output.outputResource(0xc00089ef20)
	/home/runner/work/infracost/infracost/internal/output/output.go:284 +0x105
github.com/infracost/infracost/internal/output.outputBreakdown({0xc0008ca140, 0x4, 0xc0000a3ee0?})
	/home/runner/work/infracost/infracost/internal/output/output.go:264 +0xa5
github.com/infracost/infracost/internal/output.ToOutputFormat({0xc000299700, 0x1d, 0xc0008d42d0?})
	/home/runner/work/infracost/infracost/internal/output/output.go:347 +0x44d
main.runMain(0x27a6bc0?, 0xc000104980)
	/home/runner/work/infracost/infracost/cmd/infracost/run.go:126 +0x40b
main.breakdownCmd.func1(0xc0004e6780?, {0x22c7110?, 0x3?, 0x3?})
	/home/runner/work/infracost/infracost/cmd/infracost/breakdown.go:43 +0x12a
github.com/spf13/cobra.(*Command).execute(0xc0004e6780, {0xc000027890, 0x3, 0x3})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00001af00)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
main.Run(0x0, 0x0)
	/home/runner/work/infracost/infracost/cmd/infracost/main.go:86 +0x185
main.main()
	/home/runner/work/infracost/infracost/cmd/infracost/main.go:34 +0x1d

Environment:
Infracost v0.10.8

An unexpected error occurred. We've been notified of it and will investigate it soon. If you would like to follow-up, please copy the above output and create an issue at: https://github.com/infracost/infracost/issues/new
```

**Comment by aliscott** [comment]
@amanenk thanks for reporting this. I've just been trying to debug this issue myself as part of https://github.com/infracost/infracost/issues/1875, so it would be great if you had any more information that could help us track this down.

Are you able to try and track down which resource/module is causing this by just commenting different ones out until it doesn't happen? I'm happy to jump on a call as well to debug this if you're keen - my emails is [alistair@infracost.io](mailto:alistair@infracost.io) or ping me on the [Infracost community slack](https://infracost.io/community-chat).

**Comment by aliscott** [comment]
@amanenk I've pushed a potential fix in #1923. You can try this in your action by using a dev build of Infracost I've pushed to an S3 bucket, by temporarily updating your action to have:
```
      - name: Setup Infracost
        if: steps.changes.outputs.src == 'true' || github.event_name != 'pull_request_target'
        # uses: infracost/actions/setup@v2
        run: |
          curl -o /tmp/infracost https://infracost-public-dumps.s3.amazonaws.com/builds/1923/infracost-linux-amd64
          chmod +x /tmp/infracost
          /tmp/infracost configure set api_key ${{ secrets.INFRACOST_API_KEY }}
        with:
          api-key: ${{ secrets.INFRACOST_API_KEY }}
```

You'll then need to update any `infracost` command in your workflow to use `/tmp/infracost`.

Let me know if this fixes you issue, and if so we will get this out in the next release.

**Comment by aliscott** [comment]
@amanenk thanks for your help debugging this issue 🙏 . It is now fixed in the [v0.10.9](https://github.com/infracost/infracost/releases/tag/v0.10.9) release, so you can revert the `Setup Infracost` step changes and point to the `infracost/actions/setup@v2` action now.
